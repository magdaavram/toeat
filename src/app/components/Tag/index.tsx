import React, { InputHTMLAttributes } from 'react';
import styled from 'styled-components';

interface Props extends InputHTMLAttributes<HTMLDivElement> {
  name: string;
  active: boolean;
}

const TagContainer = styled.div((props: Props) => ({
  display: 'inline-block',
  margin: '0 0 12px 12px',
  padding: '6px 12px 6px 12px',
  borderRadius: '9px',
  backgroundColor: props.active ? 'rgba(110, 37, 52, 0.7)' : 'rgba(110, 37, 52, 0.3)',
  color: 'var(--color--beige)',
  fontSize: 'var(--font-size--small)',
  cursor: 'pointer',
}));

const Tag = (props: Props) => {
  return (
    <TagContainer {...props} onClick={() => console.log(`pressed ${props.name}`)}>
      <span>{props.name}</span>
    </TagContainer>
  );
};

export default Tag;
