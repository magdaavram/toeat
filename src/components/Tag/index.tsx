import React, { InputHTMLAttributes } from 'react';
import styled from 'styled-components';

interface IProps extends InputHTMLAttributes<HTMLDivElement> {
  name: string;
  active: boolean;
}

const TagContainer = styled.div((props: IProps) => ({
  display: 'inline-block',
  margin: '0 0 12px 12px',
  padding: '6px 12px 6px 12px',
  borderRadius: '9px',
  backgroundColor: props.active ? 'rgba(110, 37, 52, 0.7)' : 'rgba(110, 37, 52, 0.3)',
  color: 'var(--color--beige)',
  fontSize: 'var(--font-size--small)',
  cursor: 'pointer',
}));

const Tag = (props: IProps) => {
  const { name } = props;

  return (
    <TagContainer {...props} onClick={() => console.log(`pressed ${name}`)}>
      <span>{name}</span>
    </TagContainer>
  );
};

export default Tag;
