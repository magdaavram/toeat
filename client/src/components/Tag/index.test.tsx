import React from 'react';
import { create } from 'react-test-renderer';
import { shallow } from 'enzyme';
import Tag from './index';

it('should render as active', () => {
  const tree = create(<Tag name={'tomato'} active={true} />);

  expect(tree.toJSON()).toMatchSnapshot();
});

it('should render as inactive', () => {
  const tree = create(<Tag name={'tomato'} active={false} />);

  expect(tree.toJSON()).toMatchSnapshot();
});

it('should call the onClick function', () => {
  const onClick = jest.fn();
  const wrapper = shallow(<Tag name={'tomato'} active={true} onClick={onClick} />);

  wrapper.props().onClick();
  expect(onClick).toHaveBeenCalledTimes(1);
});
