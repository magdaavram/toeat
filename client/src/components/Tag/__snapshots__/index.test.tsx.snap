// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render as active 1`] = `
<div
  className="sc-AxjAm caFkQs"
  name="tomato"
>
  <span>
    tomato
  </span>
</div>
`;

exports[`should render as inactive 1`] = `
<div
  className="sc-AxjAm YZIJc"
  name="tomato"
>
  <span>
    tomato
  </span>
</div>
`;
